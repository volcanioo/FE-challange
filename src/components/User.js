import React from 'react'

import "./User.scss";

function user({name, image, scores, nextMoveFunc, isWinner}) {
  const className = isWinner ? "user user--winner" : "user";
  return (
    <>
      <section className={className}>
        <figure>
          <img src={image} alt="the user avatar" />
        </figure>
        <h2>{name}</h2>
        {isWinner ? <h3>ðŸŽ‰ WINNER ðŸŽ‰</h3> : null}
        <span>Score: {scores ? scores.reduce((x, y) => x + y) : 0}</span>
        {nextMoveFunc && isWinner === null &&
          <>
            <button
              className='user__button'
              onClick={() => nextMoveFunc()}
            >
              Roll
            </button>
          </>
        }
      </section>
    </>
  )
}

export default user