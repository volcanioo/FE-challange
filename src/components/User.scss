.user {
  width: 20%;
  padding: 10px;
  border: 3px solid #DD596E;
  margin: 10px;
  text-align: center;
  transition: all .3s;

  h2 {
    margin-bottom: 0;
  }

  &--winner {
    color: #1c1b22;
    background: #F8DB55;
    border-color: #F8DB55;
    text-align: center;

    .user__button {
      background: white;
    }
  }

  img {
    width: 100%;
    display: block;
    height: auto;
  }

  &__button {
    width: 100%;
    background: #dd596e;
    border: none;
    line-height: 34px;
    margin-top: 10px;
    cursor: pointer;

    &:hover {
      opacity: .9;
    }

    &[disabled] {
      opacity: .2;
    }
  }
}

@media only screen and (max-width: 600px) {
  .wrapper {
    flex-wrap: wrap;
  }

  .user {
    flex: 0 0 33.3%;
  }
  
  .user figure, .user h2 {
    margin: 0;
  }

  .winner-badge {
    background: white;
    padding: 10px;
    text-align: center;
    color: black;
    margin: 0px 0 10px 0;
    font-size: 14px;
  }
}
