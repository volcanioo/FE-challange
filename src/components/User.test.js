import React from 'react';
import renderer from 'react-test-renderer';
import User from './User.js';

const dummyPlayer = {
  id: 1,
  name: 'John',
  imageUrl: 'https://www.fillmurray.com/200/300',
  scores: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
}

it('snapshot check', () => {
  const component = renderer.create(
    <User
      name={dummyPlayer.name}
      image={dummyPlayer.imageUrl}
      scores={dummyPlayer.scores}
      nextMoveFunc={false}
      isWinner={false}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('checks if it shows winner badge', () => {
  const component = renderer.create(
    <User
      name={dummyPlayer.name}
      image={dummyPlayer.imageUrl}
      scores={dummyPlayer.scores}
      nextMoveFunc={false}
      isWinner={true}
    />
  );
  let tree = component.toJSON();
  expect(String(tree.children[2].children)).toEqual('ðŸŽ‰ WINNER ðŸŽ‰');
  expect(tree).toMatchSnapshot();
});