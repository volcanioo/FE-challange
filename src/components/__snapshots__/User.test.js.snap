// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if it shows winner badge 1`] = `
<section
  className="user user--winner"
>
  <figure>
    <img
      alt="the user avatar"
      src="https://www.fillmurray.com/200/300"
    />
  </figure>
  <h2>
    John
  </h2>
  <h3>
    ðŸŽ‰ WINNER ðŸŽ‰
  </h3>
  <span>
    Score: 
    55
  </span>
</section>
`;

exports[`snapshot check 1`] = `
<section
  className="user"
>
  <figure>
    <img
      alt="the user avatar"
      src="https://www.fillmurray.com/200/300"
    />
  </figure>
  <h2>
    John
  </h2>
  <span>
    Score: 
    55
  </span>
</section>
`;
